.github/workflows/end-to-end-tests.yml

```diff
@@ -19,14 +19,20 @@ on:
       - '[6-9].[0-9]'
   workflow_dispatch:
 
+# Cancels all previous workflow runs for pull requests that have not completed.
+concurrency:
+  # The concurrency group contains the workflow name and the branch name for pull requests
+  # or the commit hash for any other events.
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
```

Comment:
Since it's not really possible to have the same workflow run more than once at the same time per commit this could also be set to `true` by default.

Reply:
I went back and forth here, but I think I prefer just using `true`.